Tangem/App/Services/WalletConnect/WalletConnectService/WalletConnectV1Service.swift

```diff
@@ -6,70 +6,14 @@
 //  Copyright © 2021 Tangem AG. All rights reserved.
 //
 
-import Foundation
 import WalletConnectSwift
 import Combine
 import TangemSdk
-import BlockchainSdk
-import CryptoSwift
-import SwiftUI
-import web3swift
-
-protocol WalletConnectChecker: AnyObject {
-    var isServiceBusy: CurrentValueSubject<Bool, Never> { get }
-    func containSession(for wallet: WalletInfo) -> Bool
-}
-
-protocol WalletConnectSessionController: WalletConnectChecker {
-    var sessionsPublisher: AnyPublisher<[WalletConnectSession], Never> { get }
-    func disconnectSession(_ session: WalletConnectSession)
-    func canHandle(url: String) -> Bool
-    func handle(url: String) -> Bool
-}
-
-protocol WalletConnectHandlerDelegate: AnyObject {
-    func send(_ response: Response, for action: WalletConnectAction)
-    func sendInvalid(_ request: Request)
-    func sendReject(for request: Request, with error: Error, for action: WalletConnectAction)
-    func sendUpdate(for session: Session, with walletInfo: Session.WalletInfo)
-}
-
-protocol WalletConnectHandlerDataSource: AnyObject {
-    var cardModel: CardViewModel { get }
-    func session(for request: Request) -> WalletConnectSession?
-    func updateSession(_ session: WalletConnectSession)
-}
 
-enum WalletConnectAction: String {
-    case personalSign = "personal_sign"
-    case signTransaction = "eth_signTransaction"
-    case sendTransaction = "eth_sendTransaction"
-    case bnbSign = "bnb_sign"
-    case bnbTxConfirmation = "bnb_tx_confirmation"
-    case signTypedData = "eth_signTypedData"
-    case signTypedDataV4 = "eth_signTypedData_v4"
-    case switchChain = "wallet_switchEthereumChain"
-
-//    var shouldDisplaySuccessAlert: Bool {
-//        switch self {
-//        case .bnbTxConfirmation: return false
-//        default: return true
-//        }
-//    }
-
-    var successMessage: String {
-        switch self {
-        case .personalSign, .signTypedData, .signTypedDataV4: return "wallet_connect_message_signed".localized
-        case .signTransaction: return "wallet_connect_transaction_signed".localized
-        case .sendTransaction: return "wallet_connect_transaction_signed_and_send".localized
-        case .bnbSign: return "wallet_connect_bnb_transaction_signed".localized
-        case .bnbTxConfirmation, .switchChain: return "".localized
-        }
-    }
-}
+fileprivate typealias ExtractedWCUrl = (url: String, handleDelay: TimeInterval)
 
-class WalletConnectService: ObservableObject {
-    var isServiceBusy: CurrentValueSubject<Bool, Never> = .init(false)
+class WalletConnectV1Service {
+    var canEstablishNewSessionPublisher: CurrentValueSubject<Bool, Never> = .init(true)
```

Comment:
а почему так?) Чисто вкусовщина)
По мне так лучше )

```suggestion
    var canEstablishNewSessionPublisher = CurrentValueSubject<Bool, Never>(true)
```

Reply:
да это старый вариант, кажется раньше он вообще был `@Published`, а потом перевели на `CurrentValueSubject`, а инициализацию не поменяли. Обновлю

Reply:
обновил, поперекладывал поля дополнительно